## __PARTE 1 Carga de Datos__

In [5]:
#Lee la libreria de pandas para poder abrir el archivo y siguientes análisis
import pandas as pd
df = pd.read_csv("dataset_practiva.csv")

In [6]:
#Mostrar las primeras 10 filas 
print(df.head(10))


   ID_Cliente     Nombre       Ciudad  Edad                   Email Membresia  \
0          97   Jorge Z.       Puebla  25.0                     NaN     Plata   
1          23   Maria D.  Guadalajara  18.0    maria.d23@correo.com    Bronce   
2         114  Andres X.    Monterrey  54.0  andres.x114@correo.com       Oro   
3         139   Paola X.       Toluca  18.0   paola.x139@correo.com     Plata   
4         162  Renata F.       Toluca  26.0  renata.f162@correo.com       Oro   
5         112  Hector X.    Monterrey  61.0  hector.x112@correo.com     Plata   
6           3  Renata E.    Queretaro  22.0    renata.e3@correo.com       Oro   
7         103   Karla H.       Toluca  63.0   karla.h103@correo.com       Oro   
8          25  Hector K.  Guadalajara   NaN   hector.k25@correo.com    Bronce   
9          61  Emilio R.       Puebla  20.0   emilio.r61@correo.com    Bronce   

   Num_Compras  Total_Gastado  
0            4        1610.42  
1            0        1303.73  
2           

In [7]:
#Mostrar las ùltimas 5 filas 
print(df.tail(5))


     ID_Cliente      Nombre       Ciudad  Edad                   Email  \
255         212    Pablo D.       Puebla  37.0   pablo.d212@correo.com   
256          68  Ricardo S.  Guadalajara  23.0  ricardo.s68@correo.com   
257          26   Renata M.  Guadalajara  62.0   renata.m26@correo.com   
258         197    Paola I.       Toluca  63.0   paola.i197@correo.com   
259         176   Andres F.       Toluca  62.0  andres.f176@correo.com   

    Membresia  Num_Compras  Total_Gastado  
255     Plata            7        2670.08  
256    Bronce            2         774.50  
257    Bronce            5         979.36  
258     Plata            9        3551.58  
259    Bronce            5        1055.27  


In [8]:
#Obtner las dimensiones
print("Dimensiones del Dataset:", df.shape)


Dimensiones del Dataset: (260, 8)


In [9]:
#Mostrar las columnas
print(df.columns)

Index(['ID_Cliente', 'Nombre', 'Ciudad', 'Edad', 'Email', 'Membresia',
       'Num_Compras', 'Total_Gastado'],
      dtype='str')


In [22]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 260 entries, 0 to 259
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   ID_Cliente     260 non-null    int64  
 1   Nombre         260 non-null    str    
 2   Ciudad         260 non-null    str    
 3   Edad           245 non-null    float64
 4   Email          247 non-null    str    
 5   Membresia      260 non-null    str    
 6   Num_Compras    260 non-null    int64  
 7   Total_Gastado  260 non-null    float64
dtypes: float64(2), int64(2), str(4)
memory usage: 16.4 KB


In [23]:
df.describe()

,ID_Cliente,Edad,Num_Compras,Total_Gastado
count,260.000000,245.000000,260.000000,260.000000
mean,126.473077,40.579592,4.900000,1910.354692
std,72.944704,13.832383,4.125165,1540.873869
min,1.000000,18.000000,0.000000,303.370000
25%,63.750000,28.000000,2.000000,874.907500
50%,126.500000,42.000000,4.000000,1161.150000
75%,189.250000,51.000000,7.000000,2418.835000
max,250.000000,64.000000,16.000000,7214.180000


In [10]:
#Mostrar el tipo de datos de cada columna
print(df.dtypes)


ID_Cliente         int64
Nombre               str
Ciudad               str
Edad             float64
Email                str
Membresia            str
Num_Compras        int64
Total_Gastado    float64
dtype: object


1.¿Cuántas observaciones existen?  
   > Existen 260 observaciones y podemos deducirlo con un ´shape´, ya que nos muestra la dimensión de nuestra dataset.   

2.¿Cuántas variables existen?  
   > Existen 8 variables o columnas, que igualmente podemos obtener con un ´shape´.  

3.¿Qué representa una fila?  
> Una fila representa una observación del dataset.  

4.¿Qué representa una columna?  
> Representa una variable del dataset.  

5.¿Qué variables son numéricas?   
   > Edad, Num_Compras, Total_Gastado.  

6.¿Qué variables son categóricas?  
   > ID_Cliente, Nombre, Ciudad, Email, Membresia. 

Conclusión:
> Con las funciones anteriores podemos obtener información general del dataset, como el número de filas y columnas, los nombres de las columnas, el tipo de datos de cada columna, y un resumen estadístico de las columnas numéricas. Esto nos permite tener una visión general del contenido del dataset y nos ayuda a identificar posibles problemas o áreas de interés para un análisis más detallado.


## __PARTE 2-3 Selección__

Selecciona:  
> - Una columna  
> - Dos o más columnas  
> - Un subconjunto de filas   
> - Observaciones que cumplan la condición  

In [24]:
#Seleccionar una columna específica
print(df['Ciudad'])

0           Puebla
1      Guadalajara
2        Monterrey
3           Toluca
4           Toluca
          ...     
255         Puebla
256    Guadalajara
257    Guadalajara
258         Toluca
259         Toluca
Name: Ciudad, Length: 260, dtype: str


In [ ]:
#Seleccionar Nombre, Ciudad y Total Gastado
mas_de_una_columna = df[["Nombre", "Ciudad", "Total_Gastado"]]
print(mas_de_una_columna)


,Nombre,Edad
0,Jorge Z.,25.0
1,Maria D.,18.0
2,Andres X.,54.0
3,Paola X.,18.0
4,Renata F.,26.0
...,...,...
255,Pablo D.,37.0
256,Ricardo S.,23.0
257,Renata M.,62.0
258,Paola I.,63.0


In [26]:
#Seleccionar un subconjunto de filas 
subconjunto_filas = df[10:20]
subconjunto_filas

,ID_Cliente,Nombre,Ciudad,Edad,Email,Membresia,Num_Compras,Total_Gastado
10,110,Pablo T.,Queretaro,55.0,pablo.t110@correo.com,Bronce,0,1222.94
11,115,Camila D.,Toluca,57.0,camila.d115@correo.com,Bronce,1,924.84
12,159,Camila Y.,CDMX,35.0,camila.y159@correo.com,Plata,5,2065.86
13,205,Andres P.,CDMX,54.0,andres.p205@correo.com,Bronce,2,880.91
14,71,Hector I.,Queretaro,19.0,hector.i71@correo.com,Bronce,1,685.17
15,109,Andres C.,Queretaro,49.0,andres.c109@correo.com,Plata,9,1877.38
16,195,Emilio N.,CDMX,24.0,NaN,Oro,14,3768.55
17,130,Miguel G.,Monterrey,47.0,miguel.g130@correo.com,Plata,8,2883.08
18,158,Sergio D.,Puebla,26.0,sergio.d158@correo.com,Bronce,2,877.85
19,4,Valentina I.,Queretaro,61.0,valentina.i4@correo.com,Bronce,0,1038.78


In [28]:
#Observaciones que cumplan una condición
observaciones_condicion = df[df['Edad'] > 31]   
observaciones_condicion

,ID_Cliente,Nombre,Ciudad,Edad,Email,Membresia,Num_Compras,Total_Gastado
2,114,Andres X.,Monterrey,54.0,andres.x114@correo.com,Oro,11,5077.40
5,112,Hector X.,Monterrey,61.0,hector.x112@correo.com,Plata,9,2116.81
7,103,Karla H.,Toluca,63.0,karla.h103@correo.com,Oro,9,4713.38
10,110,Pablo T.,Queretaro,55.0,pablo.t110@correo.com,Bronce,0,1222.94
11,115,Camila D.,Toluca,57.0,camila.d115@correo.com,Bronce,1,924.84
...,...,...,...,...,...,...,...,...
252,93,Ivan Q.,Guadalajara,60.0,ivan.q93@correo.com,Bronce,3,797.29
255,212,Pablo D.,Puebla,37.0,pablo.d212@correo.com,Plata,7,2670.08
257,26,Renata M.,Guadalajara,62.0,renata.m26@correo.com,Bronce,5,979.36
258,197,Paola I.,Toluca,63.0,paola.i197@correo.com,Plata,9,3551.58


Conclusión
> Con los parámetros anteriores podemos solo seleccionar ciertas columnas o datos, para obterner mayor información de nuestro dataset. 

## __Parte 4 - Estadisticos__  

Calcular:
> - media  
> - mediana  
> - mínimo  
> - máximo   
> - suma  
> - conteo   


In [37]:
#Obteener la Media del total_gastado 
print("La media del total gastado es:", df['Total_Gastado'].mean())

La media del total gastado es: 1910.3546923076922


In [36]:
#Obteener la Mediana del total_gastado 
print("La mediana del total gastado es:", df['Total_Gastado'].median())

La mediana del total gastado es: 1161.15


In [38]:
#Obtener el mínimo del total gastado y máximo
print("El mínimo del total gastado es:", df['Total_Gastado'].min())
print("El máximo del total gastado es:", df['Total_Gastado'].max())

El mínimo del total gastado es: 303.37
El máximo del total gastado es: 7214.18


In [39]:
#La suma del total gastado  
print("La suma del total gastado es:", df['Total_Gastado'].sum())

La suma del total gastado es: 496692.22


In [40]:
#El conteo sobre la columna de ciudad
conteo_ciudad = df['Ciudad'].value_counts()
print("Conteo de observaciones por ciudad:\n", conteo_ciudad)

Conteo de observaciones por ciudad:
 Ciudad
Toluca         53
Puebla         46
Queretaro      45
Guadalajara    43
CDMX           41
Monterrey      32
Name: count, dtype: int64


1.¿Qué variable consideras más importante del dataset?  
> 
2.¿Qué información encontraste que no era evidencia únicamente observando las primeras filas?  
> 
3.¿Qué pregunta de negocio o investigación podría responddere este dataset?  
> 